core/src/main/java/com/oath/oak/OakDetachedReadKeyBufferSynced.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import java.util.function.Function;
+
+class OakDetachedReadKeyBufferSynced extends OakDetachedReadKeyBuffer {
```

Comment:
I suggest that for any buffer class (implementation and interface), there is a header comment explaining its usage.

Reply:
I added documentation to each of the detached buffer implementations.
I hope this will answer all the concerns you raised here regarding the detached buffers.